### PR TITLE
#657 Save manually filled values on rebuild

### PR DESCRIPTION
**Pull request by adrianchivu**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### CODE REVIEW

**Review by cozminvelciu — CHANGES_REQUESTED** [review]
Still needs to be done:
* on saving & closing the Excel workbook, and then reopening it, the selection still needs to be persisted
* manually entered values also need to be persisted on sheet regeneration
* manually entered values loaded from metadata (not from model) need to be colored differently (maybe same color as for yet-to-be published values)

**Comment by cozminvelciu** [comment]
I see a few problems here:
1. We should not allow the user to edit the first fixed columns in the sheet, and the headers. They should be locked/protected.
1. Not directly related to this PR, but more to the Crossview generation as a whole: Cut & copy commands affect the whole cell data - this includes the name. Meaning that if we try to cut/copy a value from one place to another, it's going to mess up the cell names. In the parameter sheet this is not a problem because the cell names are on the "Actual value" column (which is locked, along with the rest of the sheet), while the editable values are on the "Manual" column. Sadly we can't afford to have multiple columns ("Actual", "Manual", etc.) for each ParameterValueSet in the Crossview sheet.
1. In this implementation if the user manually changes a value in the Crossview sheet it will be persisted forever in the metadata. Should the metadata for manually changed values be cleared at some point?

@alexatrhea, @lxatrhea, what are your thoughts on this?

**Review by alexatrhea — APPROVED** [review]

**Comment by cozminvelciu** [comment]
TODO: prepend Crossview_ to all cell names.

**Comment by cozminvelciu** [comment]
Checkbox with label: "Persist manually entered values in the sheet" (default on)

// "Use values from datasource [when off manually entered values are persisted in the sheet]" (default false).

**Comment by cozminvelciu** [comment]
> I see a few problems here:
> 
> 1. We should not allow the user to edit the first fixed columns in the sheet, and the headers. They should be locked/protected.
> 2. Not directly related to this PR, but more to the Crossview generation as a whole: Cut & copy commands affect the whole cell data - this includes the name. Meaning that if we try to cut/copy a value from one place to another, it's going to mess up the cell names. In the parameter sheet this is not a problem because the cell names are on the "Actual value" column (which is locked, along with the rest of the sheet), while the editable values are on the "Manual" column. Sadly we can't afford to have multiple columns ("Actual", "Manual", etc.) for each ParameterValueSet in the Crossview sheet.
> 3. In this implementation if the user manually changes a value in the Crossview sheet it will be persisted forever in the metadata. Should the metadata for manually changed values be cleared at some point?

1. Added separate issue: #710 
2. Default Excel behavior, cannot really do anything to prevent this. Added related issue for name uniqueness generation: #711.
3. The crossview sheet generation dialog should have a checkbox with the label "Persist manually eneterd values in the sheet" (default on). When this is un-checked, the sheet should disregard values save in the metadata, and clear them. Added separate issue: #712.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        /// <param name="elementDefinitions">
+        /// The element definitions list that needs to be preserved <see cref="IEnumerable{ElementDefinition}" />
+        /// </param>
+        /// <param name="parameterTypes">
+        /// The parameter types list that needs to be preserved <see cref="IEnumerable{ParameterType}" />
+        /// </param>
+        /// <param name="manuallySavedValues">
+        /// The cell name/value key/value pai that needs to be preserved <see cref="Dictionary{TKey,TValue}"/>
+        /// </param>
+        public CrossviewWorkbookData(IEnumerable<CDP4Common.EngineeringModelData.ElementDefinition> elementDefinitions, IEnumerable<CDP4Common.SiteDirectoryData.ParameterType> parameterTypes,
```
Comment (cozminvelciu) [inline CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:175]: Each parameter should be on a separate line for consistency.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            Dictionary<string, string> manuallySavedValues)
+        {
+            var preservedDefinitions = elementDefinitions.Select(elementDefinition => elementDefinition.ToDto() as ElementDefinition).ToList();
+            this.elementDefinitionData =  this.GenerateStringFromList(preservedDefinitions);
+
+            var preservedTypes = parameterTypes.Select(parameterType => parameterType.ToDto() as ParameterType).ToList();
+            this.parameterTypeData = this.GenerateStringFromList(preservedTypes);
+
+            this.cellNamesData = string.Join("|", manuallySavedValues.Keys.ToArray());
```
Comment (cozminvelciu) [inline CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:184]: We need to find a way so this serialization does not depend on the presence or absence of the hardcoded separator (which should be a constant at the top) inside the values.

For the names, we are technically ok for now, as the `|` character is not present in the prefix for each cell name, nor can it be present in any short name in the model code, but this doesn't make it ok.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+            this.cellValuesData = string.Join("|", manuallySavedValues.Values.ToArray());
```
Comment (cozminvelciu) [inline CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:186]: We need to find a way so this serialization does not depend on the presence or absence of the hardcoded separator (which should be a constant at the top) inside the values.

For the values, this doesn't work, as values are allowed to contain the `|` character.

File: CDP4CrossViewEditor/WorkbookOperator.cs [hunk 4]
```diff
@@ -42,9 +42,31 @@ namespace CDP4CrossViewEditor
...
 
+    /// <summary>
+    /// Workbook metadata that will be preserved between two running sessions
+    /// </summary>
+    public struct WorkbookMetadata
+    {
+        /// <summary>
+        /// Element definitions user selection
+        /// </summary>
+        public IEnumerable<ElementDefinition> ElementDefinitions;
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/WorkbookOperator.cs:57]: Rename to `SelectedElementDefinitions`.

File: CDP4CrossViewEditor/WorkbookOperator.cs [hunk 5]
```diff
@@ -42,9 +42,31 @@ namespace CDP4CrossViewEditor
...
+
+        /// <summary>
+        /// Parameter types user selection
+        /// </summary>
+        public IEnumerable<ParameterType> ParameterTypes;
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/WorkbookOperator.cs:62]: Rename to `SelectedParameterTypes`.

File: CDP4CrossViewEditor/WorkbookOperator.cs [hunk 6]
```diff
@@ -42,9 +42,31 @@ namespace CDP4CrossViewEditor
...
+
+        /// <summary>
+        /// Parameter sheet values changed by the user
+        /// </summary>
+        public Dictionary<string, string> ParameterValues;
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/WorkbookOperator.cs:67]: `CrossviewWorkbookdata:141` should be consistent with this.

Maybe rename both to `ManuallySavedParameterValues`.

File: CDP4CrossViewEditor/WorkbookOperator.cs [hunk 7]
```diff
@@ -62,11 +84,19 @@ public class WorkbookOperator
...
+        /// The <see cref="WorkbookMetadata"/> that is being managed by the current <see cref="WorkbookOperator"/>
+        /// </summary>
+        private readonly WorkbookMetadata workbookMetadata;
+
         /// <summary>
         /// The excel <see cref="Application"/> that is being managed by the current <see cref="WorkbookOperator"/>
         /// </summary>
         private readonly Application application;
 
+        private CrossviewSheetGenerator crossviewSheetGenerator;
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/WorkbookOperator.cs:97]: Missing XML docs.

File: CDP4CrossViewEditor/WorkbookOperator.cs [hunk 8]
```diff
@@ -62,11 +84,19 @@ public class WorkbookOperator
...
+
+
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/WorkbookOperator.cs:99]: Unneeded extra newline.

File: CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs [hunk 9]
```diff
@@ -46,14 +46,17 @@ public class WorkbookSelectionDialogResult : BaseDialogResult
         /// <param name="workbook">The <see cref="Workbook"/> that is the result of a selection.</param>
         /// <param name="workbookElements">Elements that has been selected <see cref="ElementDefinition"/></param>
         /// <param name="workbookParameterTypes">Parameter types that has been selected <see cref="ParameterType"/></param>
+        /// <param name="workbookChangedValues"></param>
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs:49]: Missing XML docs.

File: CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs [hunk 10]
```diff
@@ -46,14 +46,17 @@ public class WorkbookSelectionDialogResult : BaseDialogResult
...
         public WorkbookSelectionDialogResult(bool? res, Workbook workbook,
             IEnumerable<ElementDefinition> workbookElements,
-            IEnumerable<ParameterType> workbookParameterTypes)
+            IEnumerable<ParameterType> workbookParameterTypes,
+            Dictionary<string, string> workbookChangedValues)
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs:53]: This parameter name should be consistent with the other uses throughout the code (`WorkbookOperator:67` and ``CrossviewWorkbookdata:141``).

Maybe rename to `ManuallySavedParameterValues`.

File: CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs [hunk 11]
```diff
@@ -87,8 +87,8 @@ public bool PowerParametersEnabled
         /// </summary>
         /// <param name="iteration">Current opened iteration <see cref="Iteration"/></param>
         /// <param name="session">Current opened session <see cref="ISession"/></param>
-        public ParameterTypeSelectorViewModel(Iteration iteration, ISession session)
-            : base(iteration, session, ClassKind.ParameterType)
+        /// <param name="preservedIids">Current user selection <see cref="List{Guid}"/></param>
+        public ParameterTypeSelectorViewModel(Iteration iteration, ISession session, List<Guid> preservedIids) : base(iteration, session, ClassKind.ParameterType, preservedIids)
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs:91]: `: base...` should be on the next line.

File: CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs [hunk 12]
```diff
@@ -46,7 +46,7 @@ public sealed class ParameterTypeSelectorViewModel : ThingSelectorViewModel
         /// <summary>
         /// Hardcoded power related parameter type short names list
         /// </summary>
-        private static readonly string[] PowerParameters = { "redundancy", "P_on", "P_stby", "P_peak", "P_duty_cyc", "P_mean" };
+        private static readonly string[] PowerParameters = { "Redundancy", "P_on", "P_stby", "P_peak", "P_duty_cyc", "P_mean" };
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs:49]: We might want to make the comparison case insensitive where these are checked (if not already done).

File: CDP4CrossViewEditor/ViewModels/ElementDefinitionSelectorViewModel.cs [hunk 13]
```diff
@@ -71,8 +72,8 @@ public class ElementDefinitionSelectorViewModel : ThingSelectorViewModel
         /// </summary>
         /// <param name="iteration">Current opened iteration <see cref="Iteration"/></param>
         /// <param name="session">Current opened session <see cref="ISession"/></param>
-        public ElementDefinitionSelectorViewModel(Iteration iteration, ISession session)
-            : base(iteration, session, ClassKind.ElementDefinition)
+        /// <param name="preservedIids">Current user selection <see cref="List{Guid}"/></param>
+        public ElementDefinitionSelectorViewModel(Iteration iteration, ISession session, List<Guid> preservedIids) : base(iteration, session, ClassKind.ElementDefinition, preservedIids)
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/ElementDefinitionSelectorViewModel.cs:76]: `: base ...` should be on the next line.

File: CDP4CrossViewEditor/ViewModels/CrossViewDialogViewModel.cs [hunk 14]
```diff
@@ -171,7 +217,8 @@ private void ExecuteOk()
                 this.ParameterSelectorViewModel is ParameterTypeSelectorViewModel parameterTypeViewModel)
             {
                 this.DialogResult = new WorkbookSelectionDialogResult(true, this.SelectedWorkbook?.Workbook, elementDefinitionViewModel.ElementDefinitionTargetList.Select(e => e.Thing),
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/CrossViewDialogViewModel.cs:219]: Each argument should be on a new line for consistency.

File: CDP4CrossViewEditor/ViewModels/CrossViewDialogViewModel.cs [hunk 15]
```diff
@@ -100,6 +103,11 @@ private WorkbookRowViewModel SelectedWorkbook
             set => this.RaiseAndSetIfChanged(ref this.selectedWorkbook, value);
         }
 
+        /// <summary>
+        ///
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/CrossViewDialogViewModel.cs:107]: Missing XML docs.

File: CDP4CrossViewEditor/ViewModels/CrossViewDialogViewModel.cs [hunk 16]
```diff
@@ -100,6 +103,11 @@ private WorkbookRowViewModel SelectedWorkbook
...
+        /// </summary>
+        private Dictionary<string, string> ManuallyFilledValues = new Dictionary<string, string>();
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/CrossViewDialogViewModel.cs:109]: See other comments on MR about the name consistency for this dictionary present in 3/4 separate locations.

File: CDP4CrossViewEditor/Generator/CrossviewSheetGenerator.cs [hunk 17]
```diff
@@ -120,14 +125,8 @@ public CrossviewSheetGenerator(ISession session, Iteration iteration, Participan
...
         /// </param>
-        /// <param name="elementDefinitions">
-        /// Selected element definition list
-        /// </param>
-        /// <param name="parameterTypes">
-        /// Selected parameter types list
-        /// </param>
-        public void Rebuild(Application application, Workbook workbook, IEnumerable<ElementDefinition> elementDefinitions,
-            IEnumerable<ParameterType> parameterTypes)
+        /// <param name="workbookMetadata"></param>
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/Generator/CrossviewSheetGenerator.cs:128]: Missing XML docs.

**Review by cozminvelciu — CHANGES_REQUESTED** [review]

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs (hunk 2 above)
Comment (adrianchivu) [reply CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:184]: Added a constant for the moment.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs (hunk 3 above)
Comment (adrianchivu) [reply CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:186]: Added a constant for the moment.

File: CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs (hunk 12 above)
Comment (adrianchivu) [reply CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs:49]: Correct.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs (hunk 3 above)
Comment (adrianchivu) [reply CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs]: I will try with XmlSerializer.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs (hunk 2 above)
Comment (adrianchivu) [reply CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs]: I will try with XmlSerializer.

**Review by lxatrhea — APPROVED** [review]

File: CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs [hunk 18]
```diff
@@ -46,7 +46,7 @@ public sealed class ParameterTypeSelectorViewModel : ThingSelectorViewModel
         /// <summary>
         /// Hardcoded power related parameter type short names list
         /// </summary>
-        private static readonly string[] PowerParameters = { "Redundancy", "P_on", "P_stby", "P_peak", "P_duty_cyc", "P_mean" };
+        private static readonly string[] PowerParameters = { "redundancy", "P_on", "P_stby", "P_peak", "P_duty_cyc", "P_mean" };
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs]: Additionally, the "add power parameter types" toggle does not add them in the proper order (see [this comment](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues/658#issuecomment-756666958)).

Comment (cozminvelciu) [reply CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs]: In the `ExecuteMovePowerToTarget`/`ExecuteMovePowerToSource` methods:
```
var powerParameterTypes = this.ParameterTypeSourceList
    .Where(row => PowerParameters.Contains(row.Thing.ShortName));
```
needs to be replaced with
```
var powerParameterTypes = PowerParameters
    .Select(shortName => this.ParameterTypeSourceList.FirstOrDefault(row => string.Equals(row.Thing.ShortName, shortName, StringComparison.InvariantCultureIgnoreCase)))
    .Where(row => row != null);
```

Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs]: This should be added and handled as a separate issue: I'm also having problems with the "add power parameter types" toggle:
1. after adding with the toggle, the clear ("eraser" button) does not reset toggle
1. trying to add again afterwards results in the toggle button doing nothing

File: CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs [hunk 19]
```diff
@@ -42,21 +42,33 @@ public class WorkbookSelectionDialogResult : BaseDialogResult
...
-        /// <param name="workbook">The <see cref="Workbook"/> that is the result of a selection.</param>
-        /// <param name="workbookElements">Elements that has been selected <see cref="ElementDefinition"/></param>
-        /// <param name="workbookParameterTypes">Parameter types that has been selected <see cref="ParameterType"/></param>
-        /// <param name="workbookChangedValues"></param>
-        public WorkbookSelectionDialogResult(bool? res, Workbook workbook,
-            IEnumerable<ElementDefinition> workbookElements,
-            IEnumerable<ParameterType> workbookParameterTypes,
-            Dictionary<string, string> workbookChangedValues)
+        /// <param name="res">
+        /// The result of the dialog
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs:46]: Missing trailing dot.

File: CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs [hunk 20]
```diff
@@ -42,21 +42,33 @@ public class WorkbookSelectionDialogResult : BaseDialogResult
...
+        /// </param>
+        /// <param name="workbook">
+        /// The <see cref="Workbook"/> that is the result of a selection.
+        /// </param>
+        /// <param name="manuallySavedElementsDefinitionValues">
+        /// Elements that has been selected <see cref="ElementDefinition"/>
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs:52]: Missing trailing dot.

File: CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs [hunk 21]
```diff
@@ -42,21 +42,33 @@ public class WorkbookSelectionDialogResult : BaseDialogResult
...
+        /// </param>
+        /// <param name="manuallySavedParameterTypesValues">
+        /// Parameter types that has been selected <see cref="ParameterType"/>
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs:55]: Missing trailing dot.

File: CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs [hunk 22]
```diff
@@ -42,21 +42,33 @@ public class WorkbookSelectionDialogResult : BaseDialogResult
...
+        /// </param>
+        /// <param name="manuallySavedChangedValues">
+        /// Manually saved parameter sheet values
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/ViewModels/WorkbookSelectionDialogResult.cs:58]: Missing trailing dot.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs [hunk 23]
```diff
@@ -43,96 +46,96 @@ namespace CDP4OfficeInfrastructure.OfficeDal
...
-            set => this.cellValuesData = value.Value;
+            set => this.cellValues = value.Value;
         }
 
         /// <summary>
         /// Gets the <see cref="IEnumerable{ElementDefinition}"/> instances.
         /// </summary>
         [XmlIgnore]
-        public IEnumerable<Thing> ElementDefinitionList => this.Serializer.Deserialize(this.GenerateStreamFromString(this.ElementDefinitionData.Value));
+        public IEnumerable<Thing> ManuallySavedElementDefinitionValues => this.Serializer.Deserialize(this.GenerateStreamFromString(this.SelectedElementDefinitions.Value));
```
Comment (cozminvelciu) [inline CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:132]: Rename to `SavedElementDefinitions`: they are not "manually added values" as in the case of the cells.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs [hunk 24]
```diff
@@ -43,96 +46,96 @@ namespace CDP4OfficeInfrastructure.OfficeDal
...
 
         /// <summary>
         /// Gets the <see cref="IEnumerable{ParameterType}"/> instances.
         /// </summary>
         [XmlIgnore]
-        public IEnumerable<Thing> ParameterTypeList => this.Serializer.Deserialize(this.GenerateStreamFromString(this.ParameterTypeData.Value));
+        public IEnumerable<Thing> ManuallySavedParameterValues => this.Serializer.Deserialize(this.GenerateStreamFromString(this.SelectedParameterTypes.Value));
```
Comment (cozminvelciu) [inline CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:138]: Rename to `SavedParameterTypes`: they are not "manually added values" as in the case of the cells.

File: CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs [hunk 25]
```diff
@@ -170,20 +176,22 @@ public CrossviewWorkbookData()
...
         /// </param>
-        public CrossviewWorkbookData(IEnumerable<CDP4Common.EngineeringModelData.ElementDefinition> elementDefinitions, IEnumerable<CDP4Common.SiteDirectoryData.ParameterType> parameterTypes,
+        public CrossviewWorkbookData(
+            IEnumerable<CDP4Common.EngineeringModelData.ElementDefinition> elementDefinitions,
+            IEnumerable<CDP4Common.SiteDirectoryData.ParameterType> parameterTypes,
             Dictionary<string, string> manuallySavedValues)
         {
             var preservedDefinitions = elementDefinitions.Select(elementDefinition => elementDefinition.ToDto() as ElementDefinition).ToList();
-            this.elementDefinitionData =  this.GenerateStringFromList(preservedDefinitions);
+            this.selectedElementDefinitions =  this.GenerateStringFromList(preservedDefinitions);
```
Comment (cozminvelciu) [inline CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs:187]: Extra whitespace after `=` sign.

**Review by cozminvelciu — CHANGES_REQUESTED** [review]

File: CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs (hunk 18 above)
Comment (adrianchivu) [reply CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs]: @cozminvelciu I've added a new issue: #719

Comment (adrianchivu) [reply CDP4CrossViewEditor/ViewModels/ParameterTypeSelectorViewModel.cs]: a) This was done on this PR #714
b) Changed.

**Review by cozminvelciu — APPROVED** [review]

**Pull request merged**